sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1271,10 +1271,11 @@ class DataFrame private[sql](
    * @since 1.6.0
    */
   @scala.annotation.varargs
-  def drop(colNames: String*): DataFrame = {
+  def drop(colName: String, colNames: String*): DataFrame = {
```

Comment:
Just FYI, I am not too sure about all the versions but at least this works fine at Java 7 and Scala 2.10.4.

- Scala
```scala
class TestCompatibility {
  def test(a: String): Unit = {
    test(a, Seq(): _ *)
  }
  @varargs
  def test(a: String, b: String*): Unit = {
    (a +: b).foreach(println)
  }
  def test(a: Int): Unit = {
    test(Seq(a) : _ *)
  }
  @varargs
  def test(a: Int*): Unit = {
    a.foreach(println)
  }
}
```

- Java
```java
public class Test {
    public static void main(String[] args) {
        new TestCompatibility().test("a");
        new TestCompatibility().test("a", "b");
        new TestCompatibility().test("a", "b", "c");
        new TestCompatibility().test(1);
        new TestCompatibility().test(1, 2);
        new TestCompatibility().test(1, 2, 3);
    }
}
```